json/src/de.rs

```diff
@@ -828,3 +829,53 @@ pub fn from_str<T>(s: &str) -> Result<T>
 {
     from_slice(s.as_bytes())
 }
+
+/// Iterator over JSON values
+pub struct JSONStream<T, Iter>
+    where Iter: Iterator<Item=io::Result<u8>>,
+          T: de::Deserialize
+{
+    deser: Deserializer<Iter>,
+    _marker: PhantomData<T>,
+}
+
+impl <T, Iter>JSONStream<T, Iter>
+    where Iter:Iterator<Item=io::Result<u8>>,
+          T: de::Deserialize {
+    fn new(i: Iter) -> JSONStream<T, Iter> {
```

Comment:
There is `pub struct Deserializer` and not public `impl<Iter> Deserializer<Iter>`, so I've followed this pattern. Should I change it to public?